core/src/main/java/nu/yona/server/goals/service/GoalDTO.java

```diff
@@ -56,6 +54,11 @@ public boolean isMandatory()
 
 	public abstract Goal createGoalEntity();
 
+	protected static UUID determineActivityCategoryID(String activityCategoryUrl)
+	{
+		return UUID.fromString(activityCategoryUrl.substring(activityCategoryUrl.lastIndexOf('/') + 1));
```

Comment:
Officially, we should ask this to the controller... Or rather the controller should already substitute the URL with an ID in the goal DTO when executing the POST.